crypto/evp/m_sigver.c

```diff
@@ -191,6 +191,11 @@ static int do_sigver_init(EVP_MD_CTX *ctx, EVP_PKEY_CTX **pctx,
              */
             ctx->digest = ctx->reqdigest = ctx->fetched_digest =
                 EVP_MD_fetch(locpctx->libctx, mdname, props);
+            if (ctx->digest == NULL) {
+                ctx->digest = ctx->reqdigest = EVP_get_digestbyname(mdname);
+                if (ctx->digest)
+                    ERR_pop_to_mark();
```

Comment:
Doesn't this just have the effect of clearing all errors? No mark has been set (the previous one we had was cleared on line 149).

Reply:
Fair point. Hopefully fixed.

Reply:
This still does not look right - We should start with ERR_set_mark before the EVP_MD_fetch. If that fails, we do legacy EVP_get_digestbyname. If that passes we do ERR_pop_to_mark to clear the error(s) from the fetch. In all the other cases we call ERR_clear_last_mark so errors from both regular and legacy calls are preserved. So this needs more code reorg.

Reply:
Do you describe the current code or the ideal one? Currently, the code is written in this way...

Reply:
Never mind, I got your point. Will fix.

Reply:
Fix pushed.